zigpy/quirks/v2/__init__.py

```diff
@@ -1102,14 +1103,19 @@ def prevent_default_entity_creation(
         *,
         endpoint_id: int | None = None,
         cluster_id: int | None = None,
+        cluster_type: ClusterType | None = None,
         unique_id_suffix: str | None = None,
         function: Callable[Any, bool] | None = None,
     ) -> QuirkBuilder:
         """Do not create default entities."""
+        if cluster_id is not None and cluster_type is None:
+            cluster_type = ClusterType.Server
```

Comment:
I'd have probably just put the Server cluster type as the default in the method definition, since ZHA won't care about the cluster type if there's no cluster id, but I'm also fine with this.